src/contentmanagerdelegate.cpp

```diff
@@ -285,10 +285,27 @@ void ContentManagerDelegate::handleLastColumnClicked(const QModelIndex& index, Q
 
 QSize ContentManagerDelegate::sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const
 {
-    Q_UNUSED(option);
-
-    if (index.parent().isValid()) {
-        return QSize(300, 70);
+    if (index.parent().isValid())
+    {
+        auto header = KiwixApp::instance()->getContentManager()->getView()->getView()->header();
+
+        /* Get the Rectangle from start of column1 to end of column 5. This is a
+           slight underestimate of the actual width, but it means we will produce 
+           a proper height upper bound from boundingRect and the painting process 
+           will span the width regardless of what we return in sizeHint.
+        */
+        QRect descRect(0, 0, header->length() - header->sectionSize(0), 0);
+
+        /* Based on the rectangle and text, find the best fitting size. */
+        QFontMetrics fm(option.font);
+        const QString text = index.data().toString();
+        const auto format = Qt::AlignLeft | Qt::AlignVCenter | Qt::TextWordWrap;
+        const QRect textRect = fm.boundingRect(descRect, format, text);
+        QSize textSize = textRect.size();
+
+        if (textSize.height() < 70)
+            textSize.setHeight(70);
+        return textSize;
     }
```

Comment:
This still doesn't work correctly

[Screencast from 06-07-2024 01:55:59 PM.webm](https://github.com/kiwix/kiwix-desktop/assets/11933502/7daa8b59-5fb3-4cdb-8afb-bf481276783c)

Reply:
Maybe you have to account for some padding

Reply:
Hmm either the header length or the boundingRect function is being inaccurate. I will try to find an appropriate width for the height calculation.